.github/workflows/wheels.yml

```
@@ -75,9 +75,9 @@ jobs:
           CIBW_TEST_SKIP: "*-macosx_arm64"
           MACOSX_DEPLOYMENT_TARGET: ${{ matrix.macosx_deployment_target }}
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
-          name: dist
+          name: ${{ matrix.os }}-${{ matrix.archs }}-${{ matrix.manylinux }}
```

```suggestion
          name: ${{ matrix.os }}-${{ matrix.archs }}${{ matrix.manylinux && '-' }}${{ matrix.manylinux }}
```
Would get rid of the extra trailing `-`; not sure if it's worth sacrificing a bit of readability to make the output names cleaner?

---

Could also be something like
```suggestion
          name: >-
            ${{
              format(
                '{0}-{1}{2}{3}',
                matrix.os,
                matrix.archs,
                (matrix.manylinux && '-' || ''),
                matrix.manylinux
              )
            }}
```
_(not tested)_

---

My preference would be in the middle, with
```
${{ matrix.os }}-${{ matrix.archs }}${{ matrix.manylinux && format('-{0}', matrix.manylinux) }}
```
but I have a feeling that would make no one happy.

---

> My preference would be in the middle, with [...]

Actually, that looks like the best option to me also.

---

Cool, thanks, I've pushed a commit for that.